plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergGetTableStatisticsOperations.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultiset;
+import com.google.common.io.Files;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.CountingAccessMetadata;
+import io.trino.metadata.MetadataManager;
+import io.trino.plugin.hive.metastore.Database;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.spi.security.PrincipalType;
+import io.trino.testing.LocalQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static com.google.inject.util.Modules.EMPTY_MODULE;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.sql.planner.LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.transaction.TransactionBuilder.transaction;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+
+// Cost-based optimizers' behaviors are affected by the statistics returned by the Connectors. Here is to count the getTableStatistics calls
+// when CBOs work with Iceberg Connector.
+@Test(singleThreaded = true) // counting metadata is a shared mutable state
+public class TestIcebergGetTableStatisticsOperations
+{
+    private final LocalQueryRunner queryRunner;
+    private final CountingAccessMetadata metadata;
+    private File baseDir;
+
+    public TestIcebergGetTableStatisticsOperations()
+    {
+        queryRunner = LocalQueryRunner.builder(testSessionBuilder().build())
```

Comment:
same here -- 

don't initialize test resources before test class is actually being run by the test runner
init should be done in a `@BeforeClass` method

for this, extend from `AbstractTestQueryFramework` and it will do both setup and cleanup for you.